### PR TITLE
Add more docs to foundation

### PR DESCRIPTION
**Pull request by hamelsmu**

### CODE REVIEW

**Comment by hamelsmu** [comment]
@jph00 this is ready for review

File: nbs/01_foundation.ipynb [hunk 1]
```diff
@@ -1096,6 +1096,7 @@
    "source": [
```
Comment (jph00) [inline nbs/01_foundation.ipynb:1096]: <p>Maybe mention that by default it delegates to the superclass in this case?</p>
<br/>

 _Reply via <a href='https://app.reviewnb.com/fastai/fastcore/pull/57/discussion/'>ReviewNB</a>_ <div id='ReviewNBCommentContext-DoNotDelete' style='display:none' data-state='OPEN' data-cellIndex='106'/>

Comment (jph00) [inline nbs/01_foundation.ipynb:1096]: <p>I don't love the term "monkey patch". It's not well-known outside python community. Maybe "add a list of functions or methods to an existing class"?</p>
<br/>

 _Reply via <a href='https://app.reviewnb.com/fastai/fastcore/pull/57/discussion/'>ReviewNB</a>_ <div id='ReviewNBCommentContext-DoNotDelete' style='display:none' data-state='OPEN' data-cellIndex='111'/>

Comment (jph00) [inline nbs/01_foundation.ipynb:1096]: <p>I wouldn't use "patch" in this section at all. We're just setting an attr in an instance. "patch" suggests we're changing the class.</p>
<br/>

 _Reply via <a href='https://app.reviewnb.com/fastai/fastcore/pull/57/discussion/'>ReviewNB</a>_ <div id='ReviewNBCommentContext-DoNotDelete' style='display:none' data-state='OPEN' data-cellIndex='113'/>

Comment (jph00) [inline nbs/01_foundation.ipynb:1096]: <p>Ditto re "patch"</p>
<br/>

 _Reply via <a href='https://app.reviewnb.com/fastai/fastcore/pull/57/discussion/'>ReviewNB</a>_ <div id='ReviewNBCommentContext-DoNotDelete' style='display:none' data-state='OPEN' data-cellIndex='115'/>

Comment (jph00) [inline nbs/01_foundation.ipynb:1096]: <p>No these recent ones weren't changing the class attribute. They were changing the instance attribute</p>
<br/>

 _Reply via <a href='https://app.reviewnb.com/fastai/fastcore/pull/57/discussion/'>ReviewNB</a>_ <div id='ReviewNBCommentContext-DoNotDelete' style='display:none' data-state='OPEN' data-cellIndex='121'/>

Comment (jph00) [inline nbs/01_foundation.ipynb:1096]: <p>Describe more about <em>why</em> we're doing this. i.e. so we can have 1-line func defs, and so we can choose when and whether to mix docs and code</p>
<br/>

 _Reply via <a href='https://app.reviewnb.com/fastai/fastcore/pull/57/discussion/'>ReviewNB</a>_ <div id='ReviewNBCommentContext-DoNotDelete' style='display:none' data-state='OPEN' data-cellIndex='130'/>

Comment (jph00) [inline nbs/01_foundation.ipynb:1096]: <p>I'd rather remove the <code>help</code>  call. I think the output is kinda confusing</p>
<br/>

 _Reply via <a href='https://app.reviewnb.com/fastai/fastcore/pull/57/discussion/'>ReviewNB</a>_ <div id='ReviewNBCommentContext-DoNotDelete' style='display:none' data-state='OPEN' data-cellIndex='134'/>

**Comment by hamelsmu** [comment]
Thanks, I went through and addressed all of your feedback.  I also took the opportunity to proofread carefully and improve some additional language and grammar nits I found.  

@jph00 this is ready for review again

**Comment by hamelsmu** [comment]
I made another change in that I corrected the size of the screenshots manually.  on fastcore.fast.ai the screenshots currently look like this:

image

After this correction they look like this:

image

Note that I also took off the caption as that seemed unnecessary.

**Comment by hamelsmu** [comment]
@jph00 in addition to addressing your comments I made the following updates to this PR:

1. Edited language and grammar nits I found
2. Removed the width parameter for screenshots per discussion
3. I added docs for `bind`.

Ready for review

**Pull request merged**